libraries/net/node.cpp

```diff
@@ -3165,30 +3189,48 @@ namespace graphene { namespace net { namespace detail {
+            ret_val = originating_peer->get_socket().remote_endpoint();
+            if (originating_peer->inbound_port != originating_peer->outbound_port)
+               ret_val = fc::ip::endpoint(ret_val.get_address(), originating_peer->inbound_port);
+         }
+         ret_val = message.endpoint_to_check;
+         return ret_val;
+      }
+
     void node_impl::on_check_firewall_message(peer_connection* originating_peer,
                                               const check_firewall_message& check_firewall_message_received)
     {
       VERIFY_CORRECT_THREAD();
 
+      const fc::ip::endpoint endpoint_to_check = get_endpoint_to_check(originating_peer, check_firewall_message_received );
+
       if (check_firewall_message_received.node_id == node_id_t() &&
-          check_firewall_message_received.endpoint_to_check == fc::ip::endpoint())
-      {
-        // originating_peer is asking us to test whether it is firewalled
-        // we're not going to try to connect back to the originating peer directly,
-        // instead, we're going to coordinate requests by asking some of our peers
-        // to try to connect to the originating peer, and relay the results back
-        wlog("Peer ${peer} wants us to check whether it is firewalled", ("peer", originating_peer->get_remote_endpoint()));
-        firewall_check_state_data* firewall_check_state = new firewall_check_state_data;
-        // if they are using the same inbound and outbound port, try connecting to their outbound endpoint.
-        // if they are using a different inbound port, use their outbound address but the inbound port they reported
-        fc::ip::endpoint endpoint_to_check = originating_peer->get_socket().remote_endpoint();
-        if (originating_peer->inbound_port != originating_peer->outbound_port)
-          endpoint_to_check = fc::ip::endpoint(endpoint_to_check.get_address(), originating_peer->inbound_port);
-        firewall_check_state->endpoint_to_test = endpoint_to_check;
-        firewall_check_state->expected_node_id = originating_peer->node_id;
-        firewall_check_state->requesting_peer = originating_peer->node_id;
-
-        forward_firewall_check_to_next_available_peer(firewall_check_state);
+            check_firewall_message_received.endpoint_to_check == fc::ip::endpoint())
+      {
+         // originating_peer is asking us to test whether it is firewalled
+         // do not bother if this peer should not be advertised.
+         if ( _address_builder != nullptr
+               && !_address_builder->should_advertise( endpoint_to_check ))
+            return;
+         // we're not going to try to connect back to the originating peer directly,
+         // instead, we're going to coordinate requests by asking some of our peers
+         // to try to connect to the originating peer, and relay the results back
+         wlog("Peer ${peer} wants us to check whether it is firewalled", ("peer", originating_peer->get_remote_endpoint()));
```

Comment:
This log level is a bit too high. Also, never noticed this message in the log files. I guess the firewall checking code is actually unused, probably due to a bug or setting somewhere.